Update the description for identity_enabled to be clearer

Sacul0457 opened the pull request:
Discussion from the ddevs server: https://discord.com/channels/613425648685547541/1130595287078015027/1397115430438440981

Basically, it was noted that identity_enabled mostly returns `null` instead of `false`. 
This PR basically adds that it rarely returns `false`.

Sacul0457 committed:
Update identity_enabled description to be clearer

Sacul0457 committed:
Update user.mdx

Sacul0457 committed:
Update user.mdx

markmandel commented:
Apologies - thought this was just a comma change. Got approval happy.

> This rarely returns ``false``.

Unfortunately this doesn't explain anything better, just that's it's rarely false -- which I'm not sure why it's relevant, or makes things clearer. At the very least it's should be a "This rarely returns false because...." or what I think is likely better is deeper explanation of why a user wouldn't be displaying primary guild's server tag.

Sacul0457 committed:
Update user.mdx

I believe a user leaving a guild also returns ``false``, I could it add it but it may be really lengthy then.

Also, let me know if what I provided have exceptions or are just entirely wrong.

Sacul0457 committed:
Update user.mdx

markmandel commented:
Sweet - thanks for all the research! This looks good to me, just triple checking it with eng before approving and merging.

Zoddo commented on docs/resources/user.mdx:
```diff
@@ -119,12 +119,12 @@ Premium types denote the level of premium a user has. Visit the [Nitro](https://
+| Field             | Type       | Description                                                                                                                                                                                                                  |
+|-------------------|------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| identity_guild_id | ?snowflake | the id of the user's primary guild                                                                                                                                                                                           |
+| identity_enabled  | ?boolean   | whether the user is displaying the primary guild's server tag. This can be `null` if the system clears the identity, e.g. the server no longer supports tags. This can be `false`, e.g. the user manually removes their tag. |
```
```suggestion
| identity_enabled  | ?boolean   | whether the user is displaying the primary guild's server tag. This can be `null` if the system clears the identity, e.g. the server no longer supports tags. This will be `false` if the user manually removes their tag.   |
```
Looks better to me

Sacul0457 replied:
> Nice change. I like it. 👍🏻
> 
> Make that tweak, and we're good to merge @Sacul0457 .

Alr, thanks!

Sacul0457 committed:
Update user.mdx
